### ISSUE TITLE
Error when trying to install Java 8 151

### ISSUE THREAD

**Issue by ghost** [antoineco/aco-oracle_java#42]
Hi,

I'm getting the following error with the module:
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Could not autoload puppet/parser/functions/oracle_sso: no such file to load -- puppet_x/util at /etc/puppetlabs/code/environments/production/modules/oracle_java/manifests/download.pp:33:17 on xxxxx.

I've restarted the Puppet master but now get:
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, undefined method `empty?' for #<URI::HTTP:0x27a14f09> at /etc/puppetlabs/code/environments/production/modules/oracle_java/manifests/download.pp:33:17 on node palf05.alfresco.com

class { 'oracle_java':
    version         => '8u151',
    type            => 'jdk',
    add_alternative => true,
    ssousername     => 'auser@namehere',
    ssopassword     => 'apasswordhere',
  }

I'm using 2.9.1 of the module on Puppet 2017.1.1. The Puppet Master is running CentOS 7.

**Comment by antoineco** [comment]
Hi @itsnotify,

This is a known issue which is now (hopefully) fixed on the master branch. Would you mind trying it?

**Comment by ghost** [comment]
Sure, I'll be happy to help.

**Comment by ghost** [comment]
Hi, Yes that seems to do the trick. It's now retrieving and updating the plugin on the agents :)

The manifest is now wanting to upgrade Java. When do you think the fix will be uploaded to the forge?

**Comment by antoineco** [comment]
I'm pushing a fix for #40 today, as soon as the OP confirms it works for him I'll draft a new version.

**Comment by ghost** [comment]
Okay Thanks :)

**Comment by ajardan** [comment]
@antoineco I unfortunately get an error from the master branch:

```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: can't convert Hash into String at /etc/puppet/modules/oracle_java/manifests/download.pp:33
```

When trying to download an older version (8u72)

Any ideas how can I fix ?

**Comment by antoineco** [comment]
This issue is fixed in v2.9.2.
